File: ruby/test/iOSWebTest.rb
require 'rubygems'
require 'test/unit'
require 'appium_lib'
require 'selenium-webdriver'
require './configs.rb'
extend Test::Unit::Assertions

class IOSWebTest < Test::Unit::TestCase

  def setup
    desire_cap = DesiredCapabilities.new
    @driver = Appium::Driver.new(desire_cap.desired_caps_ios_web)
    @driver.start_driver.manage.timeouts.implicit_wait = 60
  end

  def teardown
    if !@driver.nil?
      begin
        @driver.driver_quit
      rescue Exception => e
        puts e.message
      ensure
        @driver.driver_quit
      end
    end
  end

  def test_iFixit_iOS_web
    verify_invalid_username
    verify_invalid_password
    verify_login_successfully
  end

  def verify_invalid_username
    wrong_username_msg = 'Your username is invalid!'
    puts 'Test case 1: it should return error when we input wrong username'
    login('foo', 'SuperSecretPassword!')
    assert_true(true, (get_message.include? wrong_username_msg))
  end

  def verify_invalid_password
    wrong_password_msg = 'Your password is invalid!'
    puts 'Test case 2: it should return error when we input wrong password'
    login('tomsmith', 'SuperSecretPassword')
    assert_true(true, (get_message.include? wrong_password_msg))
  end

  def verify_login_successfully
    success_msg = 'You logged into a secure area!'
    puts 'Test case 3: it should run test successfully with correct username and password'
    login('tomsmith', 'SuperSecretPassword!')
    assert_true(true, (get_message.include? success_msg))
  end

  def login(username, password)
    @driver.driver.navigate.to('http://the-internet.herokuapp.com/login')
    @driver.find_element(:id, 'username').send_keys(username)
    @driver.find_element(:id, 'password').send_keys(password)
    @driver.find_element(:xpath, "//form[@name='login']").submit
  end

  def get_message
    return @driver.find_element(:id, 'flash').text
  end
  
end

File: python/selenium/python3.6/iOSWebTest.py
from selenium import webdriver
from selenium.webdriver.remote.remote_connection import RemoteConnection
import unittest
import sys
sys.path.append('..')
from python import configs

class iOSWebTest(unittest.TestCase):

  wrong_username_msg = 'Your username is invalid!'
  wrong_password_msg = 'Your password is invalid!'
  success_msg = 'You logged into a secure area!'

  def setUp(self):
    self._command_executor = RemoteConnection(configs.kobitonServerUrl, resolve_ip=False)
    self._command_executor.set_timeout(configs.session_timeout)
    self.driver = webdriver.Remote(configs.kobitonServerUrl, configs.desired_caps_ios_web)

  def tearDown(self):
    self.driver.quit()

  def test_ios_web(self):
    self.verify_invalid_username()
    self.verify_invalid_password()
    self.verify_login_successfully()

  def verify_invalid_username(self):
    print ('should return error when we input wrong username')
    self.login('foo', 'SuperSecretPassword!')
    self.assertTrue(self.wrong_username_msg in self.get_message())

  def verify_invalid_password(self):
    print ('should return error when we input wrong password')
    self.login('tomsmith', 'SuperSecretPassword')
    self.assertTrue(self.wrong_password_msg in self.get_message())

  def verify_login_successfully(self):
    print ('should run test successfully with correct username and password')
    self.login('tomsmith', 'SuperSecretPassword!')
    self.assertTrue(self.success_msg in self.get_message())

  def login(self, username, password):
    self.driver.get('http://the-internet.herokuapp.com/login')
    userE = self.driver.find_element_by_xpath(".//*[@id='username']")
    userE.send_keys(username)
    self.driver.find_element_by_id('password').send_keys(password)
    self.driver.find_element_by_xpath("//form[@name='login']").submit()

  def get_message(self):
    return self.driver.find_element_by_id('flash').text

if __name__ == '__main__':
  suite = unittest.TestLoader().loadTestsFromTestCase(iOSWebTest)
  unittest.TextTestRunner(verbosity=2).run(suite)